src/VeeArc.Infrastructure/DependencyInjection.cs

```diff
@@ -1,12 +1,39 @@
 ﻿using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
+using VeeArc.Infrastructure.Common.Interfaces;
+using VeeArc.Infrastructure.DataBase;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using VeeArc.Application.Common.Interfaces;
+using VeeArc.Domain.Entities;
+using VeeArc.Infrastructure.DataBase.Repositories;
+using VeeArc.Infrastructure.Persistence.Interceptors;
 
 namespace VeeArc.Infrastructure;
 
 public static class DependencyInjection
 {
     public static IServiceCollection AddInfrastructure(this IServiceCollection services, IConfiguration configuration)
     {
+        if (configuration.GetValue<bool>("UseInMemory"))
+        {
+            services.AddDbContext<ApplicationDbContext>(options =>
```

Comment:
Add your db logic in private method in this file

Reply:
Moved to extension